lightning/src/ln/chan_utils.rs

```diff
@@ -281,3 +285,113 @@ pub fn build_htlc_transaction(prev_hash: &Sha256dHash, feerate_per_kw: u64, to_s
 		output: txouts,
 	}
 }
+
+#[derive(Clone)]
+pub(crate) struct LocalCommitmentTransaction {
+	tx: Transaction
+}
+impl LocalCommitmentTransaction {
+	#[cfg(test)]
+	pub fn dummy() -> Self {
+		Self { tx: Transaction {
+			version: 2,
+			input: Vec::new(),
+			output: Vec::new(),
+			lock_time: 0,
+		} }
+	}
+
+	pub fn new_from_unsigned(mut tx: Transaction, their_sig: &Signature, our_funding_key: &PublicKey, their_funding_key: &PublicKey) -> LocalCommitmentTransaction {
+		if tx.input.len() != 1 { panic!("Tried to store a commitment transaction that had input count != 1!"); }
```

Comment:
I mean this is still an internal function, so these are more to prevent invalid usage of the struct that indicates we've confused ourselves over where the state machine is. Being able to do more is something we should add to EnforcingChannelKeys, but that can come over time once we move the add_local_sig to that.